src/mam4xx/mo_setinv.hpp

```diff
@@ -90,58 +86,37 @@ void setinv_single_level(Real invariants[nfs], const Real tfld,
 } // end setinv_single_level()
 
 KOKKOS_INLINE_FUNCTION
-void setinv(const ThreadTeam &team, const ColumnView invariants[nfs],
-            const ColumnView &tfld, const ColumnView &h2ovmr,
-            const ColumnView vmr[AeroConfig::num_gas_phase_species()],
-            const ColumnView cnst_offline[num_tracer_cnst],
-            const ColumnView &pmid) {
+void setinv(const ThreadTeam &team, const View2D &invariants,
+            const ConstColumnView &tfld, const ConstColumnView &qv,
+            const View1D cnst_offline[num_tracer_cnst],
+            const ConstColumnView &pmid) {
 
   Config setinv_config_;
   constexpr int nk = mam4::nlev;
+  constexpr Real mwh2o = haero::Constants::molec_weight_h2o;
 
   Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nk), [&](int k) {
     const Real tfld_k = tfld(k);
-    const Real h2ovmr_k = h2ovmr(k);
+    const Real qv_k = qv(k);
     const Real pmid_k = pmid(k);
+
     Real invariants_k[nfs];
-    const int nspec = AeroConfig::num_gas_phase_species();
-    Real vmr_k[nspec];
-    for (int i = 0; i < nspec; ++i) {
-      vmr_k[i] = vmr[i](k);
+    for (int i = 0; i < nfs; ++i) {
+      invariants_k[i] = invariants(k, i);
     }
+
     Real cnst_offline_k[num_tracer_cnst];
     for (int i = 0; i < num_tracer_cnst; ++i) {
       cnst_offline_k[i] = cnst_offline[i](k);
     }
-    setinv_single_level(invariants_k, tfld_k, h2ovmr_k, vmr_k, pmid_k,
-                        cnst_offline_k, setinv_config_);
-    for (int i = 0; i < nfs; ++i) {
-      invariants[i](k) = invariants_k[i];
-    }
-  }); // end kokkos::parfor(k)
-} // end setinv_nlev()
 
-KOKKOS_INLINE_FUNCTION
-void setinv(const ThreadTeam &team, const ColumnView invariants[nfs],
-            const ColumnView &tfld, const ColumnView &h2ovmr, const View2D &vmr,
-            const View2D &cnst_offline, const ColumnView &pmid) {
-
-  Config setinv_config_;
-  constexpr int nk = mam4::nlev;
-
-  Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nk), [&](int k) {
-    const Real tfld_k = tfld(k);
-    const Real h2ovmr_k = h2ovmr(k);
-    const Real pmid_k = pmid(k);
-    Real invariants_k[nfs];
-    View1D vmr_k = Kokkos::subview(vmr, k, Kokkos::ALL());
-    View1D cnst_offline_k = Kokkos::subview(cnst_offline, k, Kokkos::ALL());
+    Real h2ovmr_k = conversions::vmr_from_mmr(qv_k, mwh2o);
```

Comment:
@mingxuanwupnnl @singhbalwinder . This line of code could change the values of this routine because of the mwh2o values in Haero versus E3SM.

Reply:
Are the values similar enough?

Reply:
I found the issue. I did not realize that `setinv` (column version) uses `qv` (mass mixing ratio of H2O) instead of `vmr_h2o`. The test is now passing with the previous error threshold.